### PR TITLE
add qbo to list of distributions

### PR DESCRIPTION
**Pull request by alexeadem**
This PR adds qbo, to the list of distributions:
Kubeflow  v1.7.0 and v1.8.0 + Nvidia GPU Operator Documentation

https://docs.qbo.io/#/ai_and_ml?id=kubeflow

Kubeflow + Nvidia GPU Operator + Docker-in-Docker + Cgroups v2

https://www.qbo.io/#/blog_part_2_kubeflow

### CODE REVIEW

**Comment by alexeadem** [comment]
Added the missing CLA

**Review by Arhell — COMMENTED** [review]
/ok-to-test

**Review by hbelmiro — APPROVED** [review]

File: content/en/docs/started/installing-kubeflow.md
Comment (thesuperzapper) [inline content/en/docs/started/installing-kubeflow.md:166]: Is this the best link for people to learn about your tool?

**Review by thesuperzapper — CHANGES_REQUESTED** [review]
Seems reasonable, I have made some comments.

File: content/en/docs/started/installing-kubeflow.md
Comment (alexeadem) [reply content/en/docs/started/installing-kubeflow.md:166]: Yes that is the best link to learn about Qbo and Kubeflow

File: content/en/docs/started/installing-kubeflow.md
Comment (alexeadem) [reply content/en/docs/started/installing-kubeflow.md:163]: Qbo is a company with a free Community Edition offering that is open to the public

File: content/en/docs/started/installing-kubeflow.md
Comment (alexeadem) [reply content/en/docs/started/installing-kubeflow.md:169]: @thesuperzapper here is the changelog for QBO CE. 
https://github.com/alexeadem/qbo-ce/blob/main/CHANGELOG.md

File: content/en/docs/started/installing-kubeflow.md
Comment (andreyvelich) [inline content/en/docs/started/installing-kubeflow.md:169]: @alexeadem Please add your distribution version to the layouts, so you can add yourself to the OWNERs files and control distribution Kubeflow version when you become a Kubeflow member.
You check this example:
- DeployKF Version: https://github.com/kubeflow/website/blob/master/layouts/shortcodes/deploykf/latest-version.html
- Reference in the table: {{% deploykf/latest-version %}}

**Review by andreyvelich — COMMENTED** [review]
We just merged this PR: https://github.com/kubeflow/website/pull/3693.
@alexeadem Please rebase your PR and we can your distribution to the list.

File: content/en/docs/started/installing-kubeflow.md
Comment (alexeadem) [reply content/en/docs/started/installing-kubeflow.md]: @andreyvelich rebased, added qbo to the new table, added OWNERs and reference in table. Let me know if that works. Thanks

File: content/en/docs/started/installing-kubeflow.md
Comment (andreyvelich) [inline content/en/docs/started/installing-kubeflow.md:25]: Why do we need this change ?

File: content/en/docs/started/installing-kubeflow.md
Comment (andreyvelich) [inline content/en/docs/started/installing-kubeflow.md:161]: Do you have maintainer name for this distribution of QBO is the company name ?

**Review by andreyvelich — COMMENTED** [review]
Thanks for the update @alexeadem!
/assign @kubeflow/kubeflow-steering-committee @juliusvonkohout @thesuperzapper

**Comment by andreyvelich** [comment]
You need to sign all of your commits to pass DCO

File: content/en/docs/started/installing-kubeflow.md
Comment (alexeadem) [reply content/en/docs/started/installing-kubeflow.md:161]: QBO is the Company name or maintainer.

File: content/en/docs/started/installing-kubeflow.md
Comment (alexeadem) [reply content/en/docs/started/installing-kubeflow.md:25]: My bad vscode expanded the menu via a plugin. Let me remove it.

**Comment by andreyvelich** [comment]
I think, you should add yourself to the Kubeflow Org to be listed in the Owners files: https://github.com/kubeflow/internal-acls?tab=readme-ov-file#joining-kubeflow-github-organization
Otherwise, you can remove Owners files for now.

**Comment by thesuperzapper** [comment]
@andreyvelich if he meets the pull request and sponsorship requirements, let's add him.

Otherwise, we really can't just be adding people, as it does give them permissions.

**Comment by alexeadem** [comment]
> I think, you should add yourself to the Kubeflow Org to be listed in the Owners files: https://github.com/kubeflow/internal-acls?tab=readme-ov-file#joining-kubeflow-github-organization Otherwise, you can remove Owners files for now.

Ok I removed OWNERS file for now.

**Comment by andreyvelich** [comment]
@alexeadem Please can you sign your commits for DCO ?

**Comment by james-jwu** [comment]
/lgtm
But waiting for DCO to be signed.

**Comment by jbottum** [comment]
Is QBO the name of the company ?  I cannot find it in an internet search.

**Comment by alexeadem** [comment]
> Is QBO the name of the company ? I cannot find it in an internet search.

https://bizfileonline.sos.ca.gov/search/business

QBO LLC (202031010524)

**Comment by alexeadem** [comment]
> @alexeadem Please can you sign your commits for DCO ?

just did.

**Comment by thesuperzapper** [comment]
@alexeadem while we're halfway through transferring to the DCO, the repo is still configured to require the Google CLA to be signed.

Are you able to sign it?

**Comment by alexeadem** [comment]
> @alexeadem while we're halfway through transferring to the DCO, the repo is still configured to require the Google CLA to be signed.
> 
> Are you able to sign it?

@thesuperzapper Done. CLA shows ok now.

**Comment by thesuperzapper** [comment]
@jbottum I am not sure its correct to call this a "packaged distribution", from my look at https://docs.qbo.io/#/qke?id=kubeflow (which is the "website link"), it just looks like a guide to install the `kubeflow/manifests`:

Kubernetes-Engine-QKE-

File: content/en/docs/started/installing-kubeflow.md
Comment (thesuperzapper) [inline content/en/docs/started/installing-kubeflow.md:164]: This is unreasonably long, how about we use this:

```suggestion
          QBO Kubernetes Engine
```

File: layouts/shortcodes/qbo/latest-version.html
Comment (thesuperzapper) [inline layouts/shortcodes/qbo/latest-version.html:1]: [Your guide](https://docs.qbo.io/#/qke?id=kubeflow) only explain how to use `kubeflow/manifests` for `1.7.0`:

```suggestion
1.7.0
```

**Review by thesuperzapper — CHANGES_REQUESTED** [review]
I think we should create a table under the "Raw Kubeflow Manifests" section for guides like this one (which install raw `kubeflow/manifests`), rather than including them under "Packaged Distributions".

That will make it more clear to users what they are actually getting.

~~_PS: the "Nutanix" distribution is also just a brief install guide for `kubeflow/manifests`, so we should probably move it under this "Raw Kubeflow Manifests" table also._~~

__EDIT:__ actually, I take that back, Nutaix has their own fork of the manifests, I just missed it (https://github.com/nutanix/kubeflow-manifests), so this would be the first "guide".

**Comment by alexeadem** [comment]
> I think we should create a table under the "Raw Kubeflow Manifests" section for guides like this one (which install raw `kubeflow/manifests`), rather than including them under "Packaged Distributions".
> 
> That will make it more clear to users what they are actually getting.
> 
> _PS: the "Nutanix" distribution is also just a brief install guide for `kubeflow/manifests`, so we should probably move it under this "Raw Kubeflow Manifests" table also._

That will apply also for IBM and probably other Kubernetes distributions.  IKS, AKS, GKE (and QKE) are first and foremost Kubernetes distros that support Kubeflow installation by providing guidance on how to deploy them in their distros. Specifically for `qbo`  the goal is to support upstream installation methods with little to no modification and provide metal performance in Kubernetes. 

I believe the differentiation should be between packaged solutions whose main purpose is deploying kubeflow independent of the platforms (k8s or other), and Kubernetes distributions that support  kubeflow upstream or other.

**Comment by thesuperzapper** [comment]
> That will apply also for IBM and probably other Kubernetes distributions. IKS, AKS, GKE (and QKE) are first and foremost Kubernetes distros that support Kubeflow installation by providing guidance on how to deploy them in their distros (or specific packages if they are unable to deploy directly from upstream). Specifically for `qbo` the goal is to support upstream installation methods with little to no modification and provide metal performance in Kubernetes.
> 
> I believe the differentiation should be between packaged solutions whose main purpose is deploying kubeflow independent of the platforms (k8s or other), and Kubernetes distributions that support kubeflow upstream or other. For Kubernete distros I would make also [CNCF conformance](https://www.cncf.io/training/certification/software-conformance/) part of the requirement

@alexeadem While I agree that distinguishing generic vs specific "packaged distributions" would be a good idea and would help the user, it does not change the fact that the QBO guide doesn't really fit the definition of a "packaged distribution". 

For reference, all other distributions maintain a fork of `kubeflow/manifests` (however minimal it might be).

Either way, I have raised a PR to help you get a presence on the website by creating a new PR to create the "platform-specific guides" section and adding QBO:

- https://github.com/kubeflow/website/pull/3700

**Comment by thesuperzapper** [comment]
/hold

Pending discussion of alternative PR https://github.com/kubeflow/website/pull/3700

**Comment by alexeadem** [comment]
> > That will apply also for IBM and probably other Kubernetes distributions. IKS, AKS, GKE (and QKE) are first and foremost Kubernetes distros that support Kubeflow installation by providing guidance on how to deploy them in their distros (or specific packages if they are unable to deploy directly from upstream). Specifically for `qbo` the goal is to support upstream installation methods with little to no modification and provide metal performance in Kubernetes.
> > I believe the differentiation should be between packaged solutions whose main purpose is deploying kubeflow independent of the platforms (k8s or other), and Kubernetes distributions that support kubeflow upstream or other. For Kubernete distros I would make also [CNCF conformance](https://www.cncf.io/training/certification/software-conformance/) part of the requirement
> 
> @alexeadem While I agree that distinguishing generic vs specific "packaged distributions" would be a good idea and would help the user, it does not change the fact that the QBO guide doesn't really fit the definition of a "packaged distribution".
> 
> For reference, all other distributions maintain a fork of `kubeflow/manifests` (however minimal it might be).
> 
> Either way, I have raised a PR to help you get a presence on the website by creating a new PR to create the "platform-specific guides" section and adding QBO:
> 
> * [add QBO and "raw manifests" guides table #3700](https://github.com/kubeflow/website/pull/3700)

@thesuperzapper Got it. I like the idea of the raw manifests. Looking forward to seeing qbo there. Alternately I can maintain a minimal kubeflow/manifests as well in the future. Thanks for the help.

File: content/en/docs/started/installing-kubeflow.md
Comment (alexeadem) [reply content/en/docs/started/installing-kubeflow.md:164]: Van we do this `QBO Kubernetes Engine (QKE)` instead?

File: layouts/shortcodes/qbo/latest-version.html
Comment (alexeadem) [reply layouts/shortcodes/qbo/latest-version.html:1]: qbo has been tested with version 1.8.0 using the same guide. 

Prerequisites and dependencies for 1.8.0 are listed here:

https://docs.qbo.io/#/qke?id=kubeflow-v180-with-nvidia-gpu-support

qbot which is the `bot` doing the automatic deployment for kubeflow has CLI support for kubeflow 1.8.0 

https://github.com/alexeadem/qbot/blob/eee049f544865395a83f9a616915a68581e518aa/demos/kubeflow/qbot#L69

and pulling raw 1.8.0 manifest from upstream

https://github.com/alexeadem/qbot/blob/eee049f544865395a83f9a616915a68581e518aa/demos/kubeflow/qbot#L149

**Review by andreyvelich — APPROVED** [review]
As we discussed during the last Kubeflow community call, we would like to merge this PR to give Alex and QBO community visibility and ability to participate in Kubeflow 1.8 release.
It will take time for us to discuss @thesuperzapper proposal: https://github.com/kubeflow/website/pull/3672#pullrequestreview-1947842103 and we can make the appropriate changes in the followup PRs.

Alex already gave us a demo and enough confidence that this distribution is well-maintained and tested.
Unless other KSC members or Kubeflow community members have strong objections to move this forward, I would like to merge this PR.

/lgtm
/assign @kubeflow/kubeflow-steering-committee

**Review by terrytangyuan — APPROVED** [review]
LGTM

**Comment by thesuperzapper** [comment]
I still think we should shorten the "target platform" in this PR, as it's at least twice as long as every other one in the table (and makes the table harder to read).

We can just use `QBO Kubernetes Engine` as proposed in:
 https://github.com/kubeflow/website/pull/3672#discussion_r1531441663

**Comment by alexeadem** [comment]
> I still think we should shorten the "target platform" in this PR, as it's at least twice as long as every other one in the table (and makes the table harder to read).
> 
> We can just use `QBO Kubernetes Engine` as proposed in: [#3672 (comment)](https://github.com/kubeflow/website/pull/3672#discussion_r1531441663)

@thesuperzapper do you want me to update the PR with that change?

**Comment by thesuperzapper** [comment]
@alexeadem I will unhold if you do, so it will get merged.

**Comment by andreyvelich** [comment]
> > I still think we should shorten the "target platform" in this PR, as it's at least twice as long as every other one in the table (and makes the table harder to read).
> > We can just use `QBO Kubernetes Engine` as proposed in: [#3672 (comment)](https://github.com/kubeflow/website/pull/3672#discussion_r1531441663)
> 
> @thesuperzapper do you want me to update the PR with that change?

I think, QBO Kubernetes Engine (QKE) should work similar to others: GKE, EKS, IKS, etc.
Please update it and we can merge it.

**Comment by alexeadem** [comment]
@thesuperzapper and @andreyvelich. Shortened to QBO Kubernetes Engine (QKE) as suggested. Thanks

**Review by andreyvelich — COMMENTED** [review]
Thanks for the update @alexeadem!
/lgtm
/assign @thesuperzapper

**Pull request merged**
